### PR TITLE
feat: interact etcd via gRPC before Nginx starts

### PR DESCRIPTION
**Pull request by spacewander**
Signed-off-by: spacewander <spacewanderlzx@gmail.com>

### Description

Fixes # (issue)

### Checklist

- [ ] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: apisix/cli/etcd.lua [hunk 1]
```diff
@@ -146,6 +146,241 @@ local function request(url, yaml_conf)
...
+            pkey = cfg.key
+
+            local apisix_ssl = yaml_conf.apisix.ssl
+            if apisix_ssl and apisix_ssl.ssl_trusted_certificate then
+                cafile = apisix_ssl.ssl_trusted_certificate
+            end
+        end
+
+        cmd = str_format(
+            "etcdctl --insecure-transport=false %s %s %s %s " ..
```
Comment (soulbird) [inline apisix/cli/etcd.lua:315]: So, `etcdctl` will be a dependency for APISIX to run?

Comment (spacewander) [reply apisix/cli/etcd.lua:315]: Yes. I will package it this week.

File: t/cli/test_etcd_grpc_healthcheck.sh [hunk 2]
```diff
@@ -34,6 +34,8 @@ custom_clean_up() {
 
 trap custom_clean_up EXIT
 
+export ETCD_ENABLE_GRPC_GATEWAY=false
```
Comment (tzssangglass) [inline t/cli/test_etcd_grpc_healthcheck.sh:37]: why set `ETCD_ENABLE_GRPC_GATEWAY=false` here

Comment (spacewander) [reply t/cli/test_etcd_grpc_healthcheck.sh:37]: It is used in https://github.com/apache/apisix/pull/8548/files#diff-c54ab1e62f35fd33b56cc2d14d92095ceb6dc7a652ea75403d7e09e69c4742b6

**Review by tzssangglass — APPROVED** [review]

**Review by soulbird — APPROVED** [review]

**Pull request merged**